src/middlewared/middlewared/validators.py

```diff
@@ -79,15 +79,16 @@ def __call__(self, value):
 
 
 class Match:
-    def __init__(self, pattern, flags=0, explanation=None):
+    def __init__(self, pattern, flags=0, explanation=None, empty=False):
         self.pattern = pattern
         self.flags = flags
         self.explanation = explanation
+        self.empty = empty
 
         self.regex = re.compile(pattern, flags)
 
     def __call__(self, value):
-        if value is not None and not self.regex.match(value):
+        if value is not None and not self.empty and not self.regex.match(value):
```

Comment:
When would this ever be called with a non-empty value and `empty=True`?

Reply:
`empty=True` means that the value can be empty but it does not mean that if we have a value and it's not empty we shouldn't validate it ?

Reply:
This is so frustrating. All I want to do is make the `domain` not required. Instead, I'm battling the nuances of some stupid schema validator.

Reply:
> @yocalebo you can just specify a regex `'^[a-zA-Z\.\-\0-9]*$'` (`*` instead of `+` and it'll match both empty and non-empty string)

Well that's just brilliant, that makes this change extremely simple.